gymnasium/envs/registration.py

```diff
@@ -32,63 +30,118 @@
 else:
     import importlib.metadata as metadata
 
-if sys.version_info >= (3, 8):
-    from typing import Literal
+if sys.version_info < (3, 8):
+    from typing_extensions import Protocol
 else:
-    from typing_extensions import Literal
-
-from gymnasium import Env, error, logger
+    from typing import Protocol
 
 
 ENV_ID_RE = re.compile(
     r"^(?:(?P<namespace>[\w:-]+)\/)?(?:(?P<name>[\w:.-]+?))(?:-v(?P<version>\d+))?$"
 )
 
 
-def load(name: str) -> Callable:
-    """Loads an environment with name and returns an environment creation function.
+__all__ = [
+    "EnvSpec",
+    "registry",
+    "current_namespace",
+    # Core Functions
+    "register",
+    "make",
+    "spec",
+    "pprint_registry",
+    # Minor Functions
+    "namespace",
+    "get_env_id",
+    "parse_env_id",
+    "find_highest_version",
+    "load_env",
+    "load_plugin_envs",
```

Comment:
I don't think we should export these, they're pretty much for internal use

Reply:
(from get_env_id to load_plugin_envs)

Reply:
Ok, will remove